### PR TITLE
Feat/452 better queries for stacked bar

### PR DESCRIPTION
**Pull request by cy**
~~wip~~ done!

### CODE REVIEW

**Comment by cy** [comment]
this is ready to be looked at, not sure that I should be doing so much of the data wrangling on the angular side...also @junjun-zhang is it correct to get the unique ssm-tested cases count by summing the `doc_count`s returned by the agg in the 2nd query?

**Comment by shanewilson** [comment]
aggregations should come with total counts already - you shouldn't need to sum the buckets

**Comment by alex-wilmer** [comment]
note to self 
image

**Comment by alex-wilmer** [comment]
also.. queries so fast!

**Comment by cy** [comment]
@shanewilson 
> aggregations should come with total counts already - you shouldn't need to sum the buckets

is the total count 1307 if es gives back:
```
{
  "took": 673,
  "timed_out": false,
  "_shards": {
    "total": 10,
    "successful": 10,
    "failed": 0
  },
  "hits": {
    "total": 1307,
    "max_score": 0,
    "hits": []
  },
  "aggregations": {
    "projects": {
      "doc_count_error_upper_bound": 0,
      "sum_other_doc_count": 0,
      "buckets": [
        {
          "key": "TCGA-HNSC",
          "doc_count": 389,
          "genes": {
            "doc_count": 80976,
            "my_genes": {
              "doc_count": 998,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 201
                  },
                  {
                    "key": "ENSG00000083857",
                    "doc_count": 108
                  },
                  {
                    "key": "ENSG00000181143",
                    "doc_count": 106
                  },
                  {
                    "key": "ENSG00000131018",
                    "doc_count": 90
                  },
                  {
                    "key": "ENSG00000167548",
                    "doc_count": 79
                  },
                  {
                    "key": "ENSG00000042781",
                    "doc_count": 68
                  },
                  {
                    "key": "ENSG00000054654",
                    "doc_count": 40
                  },
                  {
                    "key": "ENSG00000124721",
                    "doc_count": 40
                  },
                  {
                    "key": "ENSG00000081479",
                    "doc_count": 39
                  },
                  {
                    "key": "ENSG00000127603",
                    "doc_count": 39
                  },
                  {
                    "key": "ENSG00000055609",
                    "doc_count": 34
                  },
                  {
                    "key": "ENSG00000138688",
                    "doc_count": 34
                  },
                  {
                    "key": "ENSG00000170927",
                    "doc_count": 33
                  },
                  {
                    "key": "ENSG00000163625",
                    "doc_count": 27
                  },
                  {
                    "key": "ENSG00000117713",
                    "doc_count": 23
                  },
                  {
                    "key": "ENSG00000145113",
                    "doc_count": 17
                  },
                  {
                    "key": "ENSG00000181555",
                    "doc_count": 9
                  },
                  {
                    "key": "ENSG00000186575",
                    "doc_count": 5
                  },
                  {
                    "key": "ENSG00000198786",
                    "doc_count": 5
                  },
                  {
                    "key": "ENSG00000198886",
                    "doc_count": 1
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-KIRC",
          "doc_count": 184,
          "genes": {
            "doc_count": 15087,
            "my_genes": {
              "doc_count": 331,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 68
                  },
                  {
                    "key": "ENSG00000181555",
                    "doc_count": 42
                  },
                  {
                    "key": "ENSG00000181143",
                    "doc_count": 25
                  },
                  {
                    "key": "ENSG00000198786",
                    "doc_count": 25
                  },
                  {
                    "key": "ENSG00000081479",
                    "doc_count": 17
                  },
                  {
                    "key": "ENSG00000198886",
                    "doc_count": 16
                  },
                  {
                    "key": "ENSG00000042781",
                    "doc_count": 14
                  },
                  {
                    "key": "ENSG00000117713",
                    "doc_count": 14
                  },
                  {
                    "key": "ENSG00000055609",
                    "doc_count": 13
                  },
                  {
                    "key": "ENSG00000127603",
                    "doc_count": 12
                  },
                  {
                    "key": "ENSG00000145113",
                    "doc_count": 12
                  },
                  {
                    "key": "ENSG00000131018",
                    "doc_count": 11
                  },
                  {
                    "key": "ENSG00000170927",
                    "doc_count": 11
                  },
                  {
                    "key": "ENSG00000163625",
                    "doc_count": 10
                  },
                  {
                    "key": "ENSG00000167548",
                    "doc_count": 10
                  },
                  {
                    "key": "ENSG00000054654",
                    "doc_count": 9
                  },
                  {
                    "key": "ENSG00000083857",
                    "doc_count": 9
                  },
                  {
                    "key": "ENSG00000124721",
                    "doc_count": 5
                  },
                  {
                    "key": "ENSG00000138688",
                    "doc_count": 4
                  },
                  {
                    "key": "ENSG00000186575",
                    "doc_count": 4
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-KIRP",
          "doc_count": 184,
          "genes": {
            "doc_count": 19772,
            "my_genes": {
              "doc_count": 394,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 51
                  },
                  {
                    "key": "ENSG00000181143",
                    "doc_count": 31
                  },
                  {
                    "key": "ENSG00000055609",
                    "doc_count": 24
                  },
                  {
                    "key": "ENSG00000081479",
                    "doc_count": 22
                  },
                  {
                    "key": "ENSG00000181555",
                    "doc_count": 22
                  },
                  {
                    "key": "ENSG00000145113",
                    "doc_count": 21
                  },
                  {
                    "key": "ENSG00000138688",
                    "doc_count": 20
                  },
                  {
                    "key": "ENSG00000167548",
                    "doc_count": 19
                  },
                  {
                    "key": "ENSG00000170927",
                    "doc_count": 19
                  },
                  {
                    "key": "ENSG00000054654",
                    "doc_count": 17
                  },
                  {
                    "key": "ENSG00000083857",
                    "doc_count": 17
                  },
                  {
                    "key": "ENSG00000131018",
                    "doc_count": 17
                  },
                  {
                    "key": "ENSG00000198886",
                    "doc_count": 17
                  },
                  {
                    "key": "ENSG00000127603",
                    "doc_count": 16
                  },
                  {
                    "key": "ENSG00000042781",
                    "doc_count": 15
                  },
                  {
                    "key": "ENSG00000198786",
                    "doc_count": 15
                  },
                  {
                    "key": "ENSG00000117713",
                    "doc_count": 14
                  },
                  {
                    "key": "ENSG00000124721",
                    "doc_count": 13
                  },
                  {
                    "key": "ENSG00000163625",
                    "doc_count": 12
                  },
                  {
                    "key": "ENSG00000186575",
                    "doc_count": 12
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-GBM",
          "doc_count": 179,
          "genes": {
            "doc_count": 54484,
            "my_genes": {
              "doc_count": 426,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 75
                  },
                  {
                    "key": "ENSG00000181143",
                    "doc_count": 48
                  },
                  {
                    "key": "ENSG00000055609",
                    "doc_count": 34
                  },
                  {
                    "key": "ENSG00000131018",
                    "doc_count": 34
                  },
                  {
                    "key": "ENSG00000054654",
                    "doc_count": 28
                  },
                  {
                    "key": "ENSG00000170927",
                    "doc_count": 28
                  },
                  {
                    "key": "ENSG00000042781",
                    "doc_count": 24
                  },
                  {
                    "key": "ENSG00000127603",
                    "doc_count": 24
                  },
                  {
                    "key": "ENSG00000124721",
                    "doc_count": 22
                  },
                  {
                    "key": "ENSG00000081479",
                    "doc_count": 20
                  },
                  {
                    "key": "ENSG00000181555",
                    "doc_count": 19
                  },
                  {
                    "key": "ENSG00000138688",
                    "doc_count": 13
                  },
                  {
                    "key": "ENSG00000163625",
                    "doc_count": 11
                  },
                  {
                    "key": "ENSG00000167548",
                    "doc_count": 11
                  },
                  {
                    "key": "ENSG00000117713",
                    "doc_count": 10
                  },
                  {
                    "key": "ENSG00000198786",
                    "doc_count": 8
                  },
                  {
                    "key": "ENSG00000083857",
                    "doc_count": 5
                  },
                  {
                    "key": "ENSG00000145113",
                    "doc_count": 4
                  },
                  {
                    "key": "ENSG00000186575",
                    "doc_count": 4
                  },
                  {
                    "key": "ENSG00000198886",
                    "doc_count": 4
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-LGG",
          "doc_count": 144,
          "genes": {
            "doc_count": 16539,
            "my_genes": {
              "doc_count": 226,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 51
                  },
                  {
                    "key": "ENSG00000181143",
                    "doc_count": 32
                  },
                  {
                    "key": "ENSG00000117713",
                    "doc_count": 20
                  },
                  {
                    "key": "ENSG00000170927",
                    "doc_count": 16
                  },
                  {
                    "key": "ENSG00000081479",
                    "doc_count": 12
                  },
                  {
                    "key": "ENSG00000131018",
                    "doc_count": 11
                  },
                  {
                    "key": "ENSG00000181555",
                    "doc_count": 10
                  },
                  {
                    "key": "ENSG00000042781",
                    "doc_count": 9
                  },
                  {
                    "key": "ENSG00000167548",
                    "doc_count": 9
                  },
                  {
                    "key": "ENSG00000124721",
                    "doc_count": 8
                  },
                  {
                    "key": "ENSG00000127603",
                    "doc_count": 8
                  },
                  {
                    "key": "ENSG00000055609",
                    "doc_count": 7
                  },
                  {
                    "key": "ENSG00000083857",
                    "doc_count": 7
                  },
                  {
                    "key": "ENSG00000138688",
                    "doc_count": 7
                  },
                  {
                    "key": "ENSG00000163625",
                    "doc_count": 6
                  },
                  {
                    "key": "ENSG00000054654",
                    "doc_count": 5
                  },
                  {
                    "key": "ENSG00000145113",
                    "doc_count": 5
                  },
                  {
                    "key": "ENSG00000198886",
                    "doc_count": 2
                  },
                  {
                    "key": "ENSG00000198786",
                    "doc_count": 1
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-ESCA",
          "doc_count": 140,
          "genes": {
            "doc_count": 26090,
            "my_genes": {
              "doc_count": 329,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 68
                  },
                  {
                    "key": "ENSG00000181143",
                    "doc_count": 44
                  },
                  {
                    "key": "ENSG00000131018",
                    "doc_count": 29
                  },
                  {
                    "key": "ENSG00000167548",
                    "doc_count": 23
                  },
                  {
                    "key": "ENSG00000042781",
                    "doc_count": 20
                  },
                  {
                    "key": "ENSG00000145113",
                    "doc_count": 20
                  },
                  {
                    "key": "ENSG00000117713",
                    "doc_count": 18
                  },
                  {
                    "key": "ENSG00000055609",
                    "doc_count": 15
                  },
                  {
                    "key": "ENSG00000138688",
                    "doc_count": 12
                  },
                  {
                    "key": "ENSG00000054654",
                    "doc_count": 11
                  },
                  {
                    "key": "ENSG00000124721",
                    "doc_count": 11
                  },
                  {
                    "key": "ENSG00000127603",
                    "doc_count": 11
                  },
                  {
                    "key": "ENSG00000198786",
                    "doc_count": 10
                  },
                  {
                    "key": "ENSG00000081479",
                    "doc_count": 9
                  },
                  {
                    "key": "ENSG00000163625",
                    "doc_count": 9
                  },
                  {
                    "key": "ENSG00000083857",
                    "doc_count": 6
                  },
                  {
                    "key": "ENSG00000170927",
                    "doc_count": 5
                  },
                  {
                    "key": "ENSG00000198886",
                    "doc_count": 5
                  },
                  {
                    "key": "ENSG00000181555",
                    "doc_count": 3
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-DLBC",
          "doc_count": 37,
          "genes": {
            "doc_count": 5834,
            "my_genes": {
              "doc_count": 85,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000145113",
                    "doc_count": 15
                  },
                  {
                    "key": "ENSG00000167548",
                    "doc_count": 14
                  },
                  {
                    "key": "ENSG00000181143",
                    "doc_count": 8
                  },
                  {
                    "key": "ENSG00000042781",
                    "doc_count": 6
                  },
                  {
                    "key": "ENSG00000083857",
                    "doc_count": 6
                  },
                  {
                    "key": "ENSG00000117713",
                    "doc_count": 6
                  },
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 6
                  },
                  {
                    "key": "ENSG00000131018",
                    "doc_count": 5
                  },
                  {
                    "key": "ENSG00000081479",
                    "doc_count": 4
                  },
                  {
                    "key": "ENSG00000054654",
                    "doc_count": 3
                  },
                  {
                    "key": "ENSG00000170927",
                    "doc_count": 3
                  },
                  {
                    "key": "ENSG00000055609",
                    "doc_count": 2
                  },
                  {
                    "key": "ENSG00000198786",
                    "doc_count": 2
                  },
                  {
                    "key": "ENSG00000198886",
                    "doc_count": 2
                  },
                  {
                    "key": "ENSG00000124721",
                    "doc_count": 1
                  },
                  {
                    "key": "ENSG00000127603",
                    "doc_count": 1
                  },
                  {
                    "key": "ENSG00000163625",
                    "doc_count": 1
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-LAML",
          "doc_count": 28,
          "genes": {
            "doc_count": 4163,
            "my_genes": {
              "doc_count": 58,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 10
                  },
                  {
                    "key": "ENSG00000138688",
                    "doc_count": 6
                  },
                  {
                    "key": "ENSG00000181143",
                    "doc_count": 5
                  },
                  {
                    "key": "ENSG00000127603",
                    "doc_count": 4
                  },
                  {
                    "key": "ENSG00000163625",
                    "doc_count": 4
                  },
                  {
                    "key": "ENSG00000181555",
                    "doc_count": 4
                  },
                  {
                    "key": "ENSG00000198786",
                    "doc_count": 4
                  },
                  {
                    "key": "ENSG00000081479",
                    "doc_count": 3
                  },
                  {
                    "key": "ENSG00000042781",
                    "doc_count": 2
                  },
                  {
                    "key": "ENSG00000054654",
                    "doc_count": 2
                  },
                  {
                    "key": "ENSG00000117713",
                    "doc_count": 2
                  },
                  {
                    "key": "ENSG00000124721",
                    "doc_count": 2
                  },
                  {
                    "key": "ENSG00000145113",
                    "doc_count": 2
                  },
                  {
                    "key": "ENSG00000167548",
                    "doc_count": 2
                  },
                  {
                    "key": "ENSG00000170927",
                    "doc_count": 2
                  },
                  {
                    "key": "ENSG00000055609",
                    "doc_count": 1
                  },
                  {
                    "key": "ENSG00000083857",
                    "doc_count": 1
                  },
                  {
                    "key": "ENSG00000131018",
                    "doc_count": 1
                  },
                  {
                    "key": "ENSG00000198886",
                    "doc_count": 1
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-KICH",
          "doc_count": 22,
          "genes": {
            "doc_count": 949,
            "my_genes": {
              "doc_count": 28,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000198786",
                    "doc_count": 8
                  },
                  {
                    "key": "ENSG00000198886",
                    "doc_count": 5
                  },
                  {
                    "key": "ENSG00000083857",
                    "doc_count": 2
                  },
                  {
                    "key": "ENSG00000127603",
                    "doc_count": 2
                  },
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 2
                  },
                  {
                    "key": "ENSG00000055609",
                    "doc_count": 1
                  },
                  {
                    "key": "ENSG00000081479",
                    "doc_count": 1
                  },
                  {
                    "key": "ENSG00000131018",
                    "doc_count": 1
                  },
                  {
                    "key": "ENSG00000138688",
                    "doc_count": 1
                  },
                  {
                    "key": "ENSG00000145113",
                    "doc_count": 1
                  },
                  {
                    "key": "ENSG00000167548",
                    "doc_count": 1
                  },
                  {
                    "key": "ENSG00000170927",
                    "doc_count": 1
                  },
                  {
                    "key": "ENSG00000181143",
                    "doc_count": 1
                  },
                  {
                    "key": "ENSG00000181555",
                    "doc_count": 1
                  }
                ]
              }
            }
          }
        }
      ]
    }
  }
}
```

File: .flowconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
+[ignore]
+.*/test
+.*/node_modules
```
Comment (shanewilson) [inline .flowconfig]: /node_modules

File: package.json [hunk 2]
```diff
@@ -93,7 +94,8 @@
     "build": "gulp",
     "prebuild": "npm install",
     "ts": "gulp ts:compile",
-    "upgrade": "npm-check-updates -u && bower-update && npm test"
+    "upgrade": "npm-check-updates -u && bower-update && npm test",
+    "flow": "flow; test $? -eq 0 -o $? -eq 2"
```
Comment (shanewilson) [inline package.json]: :black_magic:

File: package.json [hunk 3]
```diff
@@ -122,5 +124,10 @@
   "jest": {
     "testEnvironment": "node"
   },
-  "babel": { "presets": ["es2015", "react" ] }
+  "babel": {
+    "presets": [
+      "es2015",
+      "react"
```
Comment (shanewilson) [inline package.json]: `@knit/socks-react`

😁

Comment (cy) [inline package.json]: socks...

Comment (shanewilson) [inline package.json]: `npm install --save-dev @knit/socks-react`

```
// .babelrc

{
  "presets": ["@knit/socks-react"]
}
```

File: .flowconfig [hunk 4]
```diff
@@ -1,5 +1,4 @@
 [ignore]
-.*/test
 .*/node_modules
```
Comment (shanewilson) [inline .flowconfig]: /node_modules

Comment (cy) [inline .flowconfig]: but if i do that then it starts failing in node_modules
```
node_modules/@oncojs/sapien/index.js:29
 29:   const root = document.querySelector(selector);
                                           ^^^^^^^^ undefined. This type is incompatible with the expected param type of
578:   querySelector(selector: string): HTMLElement;
                               ^^^^^^ string. See lib: /private/tmp/flow/flowlib_25546a75/dom.js:578

node_modules/@oncojs/sapien/index.js:42
 42:   const maxCases = Math.max(...data.map(d => d[caseCountKey]));
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ call of method `map`. Method cannot be called on
 42:   const maxCases = Math.max(...data.map(d => d[caseCountKey]));
                                    ^^^^ mixed

node_modules/@oncojs/sapien/index.js:42
 42:   const maxCases = Math.max(...data.map(d => d[caseCountKey]));
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ call of method `map`. Method cannot be called on possibly undefined value
 42:   const maxCases = Math.max(...data.map(d => d[caseCountKey]));
                                    ^^^^ undefined

node_modules/@oncojs/sapien/index.js:59
 59:     .domain(data.map(x => x[primarySiteKey]))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ call of method `map`. Method cannot be called on
 59:     .domain(data.map(x => x[primarySiteKey]))
                 ^^^^ mixed
```
etc

File: .flowconfig [hunk 5]
```diff
@@ -0,0 +1,13 @@
+[ignore]
+.*/node_modules
```
Comment (cy) [inline .flowconfig]: ok, ignored all the flow erroring modules

**Pull request merged**